vernac/obligations.ml

```diff
@@ -523,11 +523,11 @@ let declare_mutual_definition l =
       (List.map (fun x -> 
 	let subs, typ = (subst_body true x) in
         let env = Global.env () in
-        let sigma = Evd.from_env env in
+        let sigma = Evd.from_ctx x.prg_ctx in
         let term = snd (Reductionops.splay_lam_n env sigma len (EConstr.of_constr subs)) in
         let typ = snd (Reductionops.splay_prod_n env sigma len (EConstr.of_constr typ)) in
-	let term = EConstr.Unsafe.to_constr term in
-	let typ = EConstr.Unsafe.to_constr typ in
+        let term = EConstr.to_constr sigma term in
+        let typ = EConstr.to_constr sigma typ in
```

Comment:
Are `term` and `typ` guaranteed to be ground here? Otherwise you need to use the `allow_undefined_evars` flag.

Reply:
What is the status of this flag? Isn't it supposed to be removed in the released version?

Reply:
Reading the code it looks like these terms should be evar-free though.

Reply:
@ppedrot the release version will just disable the anomaly, as not to bother users; however all new code should be aware to respect the invariant, as the anomaly will stay in master.

Reply:
@ejgallego Is there an issue about this?